analysis/paladinholy/src/modules/shadowlands/legendaries/MaraadsCastRatio.tsx

```diff
@@ -20,23 +32,48 @@ class MaraadsCastRatio extends Analyzer {
       this.LOTMcast,
     );
     this.addEventListener(
-      Events.cast.spell(SPELLS.LIGHT_OF_DAWN_CAST).by(SELECTED_PLAYER),
-      this.LODcast,
+      Events.applybuff.spell(SPELLS.MARAADS_DYING_BREATH_BUFF).by(SELECTED_PLAYER),
+      this.MaraadsBuffAdded,
+    );
+    this.addEventListener(
+      // this fires every time, whether its removed by LOTM, naturally expired, or overwritten by another LOD cast (for some reason)
+      Events.removebuff.spell(SPELLS.MARAADS_DYING_BREATH_BUFF).by(SELECTED_PLAYER),
+      this.MaraadsBuffRemoved,
+    );
+    this.addEventListener(
+      // this only fires when the Maraads buff is removed by LOTM, not when it naturally expires or is overwritten.
+      Events.removebuffstack.spell(SPELLS.MARAADS_DYING_BREATH_BUFF).by(SELECTED_PLAYER),
+      this.MaraadsBuffStacksRemoved,
     );
   }
 
   LOTMcast(event: CastEvent) {
-    this.LOTMCasts += 1;
+    this.totalCasts += 1;
+    if (this.maraadsBuffed === true) {
+      this.maraadsBuffed = false;
+    } else {
+      this.unbuffedCasts += 1;
+    }
+  }
+
+  MaraadsBuffAdded(event: ApplyBuffEvent) {
+    this.maraadsBuffApplied += 1;
+    this.maraadsBuffed = true;
+  }
+
+  MaraadsBuffRemoved(event: RemoveBuffEvent) {
+    this.removals += 1;
   }
 
-  LODcast(event: CastEvent) {
-    this.LODcasts += 1;
+  MaraadsBuffStacksRemoved(event: RemoveBuffStackEvent) {
+    // if you have the same number of buff removals as buff stack removals, this will be 0, meaning you wasted 0 buffs
```

Comment:
this seems weird, but from a quick log check it looks okay actually? odd.

Reply:
Maraads is a very strange buff
When you cast LoD you get Buff applied
When you cast LoTM you get Buff Stack Lost -> Buff Removed
When you cast LoD when you already have Maraad's buff you get Buff Removed -> Buff applied (note this isn't a refresh event but rather an actual removed -> applied event combo)
When Maraad's fades naturally it just does buff removed

So Tracking RemoveBuffStackEvent = effectively used Maraad's buffs

Reply:
wild